extensions/cl_ext_buffer_device_address.asciidoc

```diff
@@ -0,0 +1,320 @@
+   for its lifetime. Each device in the context can have their own (fixed)
+   device-side address and a copy of the created buffer which are synchronized
+   implicitly by the runtime. The main difference to a default cl_mem allocation
+   in that case is that the addresses are queriable with CL_MEM_DEVICE_PTRS_EXT
+   and the per-device address is guaranteed to be the same for the entire lifetime
+   of the cl_mem.
+
+   The device addresses of sub-buffers derived from CL_MEM_DEVICE_PRIVATE_ADDRESS_EXT
+   allocated buffers can be computed by adding the sub-buffer origin to the
+   device-specific start address.
+
+|====
+
+// refError
+
+=== Section 5.5.6 - Memory Object Queries
+
+Add a new information type <<clGetMemObjectInfo, List of supported param_names table>>:
+
+[width="100%",cols="<33%,<17%,<50%",options="header"]
+|====
+| Memory Object Info | Return type | Description
+| {CL_MEM_DEVICE_PTR_EXT_anchor}
+
+include::{generated}/api/version-notes/CL_MEM_DEVICE_PTR_EXT.asciidoc[]
+  | {cl_mem_device_address_EXT_TYPE}
+      | Returns the device address for a buffer allocated with
+   CL_MEM_DEVICE_SHARED_ADDRESS_EXT. If the buffer was not created with the flag
+   or there are multiple devices in the context and the buffer address is
+   not the same for all of them, it returns CL_INVALID_MEM_OBJECT.
+
+| {CL_MEM_DEVICE_PTRS_EXT_anchor}
+include::{generated}/api/version-notes/CL_MEM_DEVICE_PTRS_EXT.asciidoc[]
+  | {cl_mem_device_address_pair_EXT_TYPE}
+      | Returns the device-address pairs for all devices in the context.
+      The per-device addresses might differ when the buffer was allocated
+      with the CL_MEM_DEVICE_PRIVATE_ADDRESS_EXT enabled.
+|====
+
+
+=== Section 5.9.2 - Setting Kernel Arguments
+
+Add a new kernel argument setter for device pointers <<setting-kernel-arguments, Section 5.9.2>>:
+
+To set a device pointer as the argument value for a specific argument of a
+kernel, call the function
+
+include::{generated}/api/protos/clSetKernelArgDevicePointerEXT.txt[]
+include::{generated}/api/version-notes/clSetKernelArgDevicePointerEXT.asciidoc[]
+
+  * _kernel_ is a valid kernel object.
+  * _arg_index_ is the argument index.
+    Arguments to the kernel are referred by indices that go from 0 for the
+    leftmost argument to _n_ - 1, where _n_ is the total number of arguments
+    declared by a kernel.
+  * _arg_value_ is the device pointer that should be used as the argument value for
+    argument specified by _arg_index_.
+    The device pointer specified is the value used by all API calls that enqueue
+    _kernel_ ({clEnqueueNDRangeKernel} and {clEnqueueTask}) until the argument
+    value is changed by a call to {clSetKernelArgDevicePointer} for _kernel_.
+    The device pointer can only be used for arguments that are declared to be a
+    pointer to `global` memory allocated with clCreateBuffer() with the
+    CL_MEM_DEVICE_SHARED_ADDRESS_EXT flag. The pointer value specified as the argument value
```

Comment:
Presumably those created with `CL_MEM_DEVICE_PRIVATE_ADDRESS_EXT` can be used too, can't they?